src/System.Linq/src/System/Linq/Reverse.cs

```diff
@@ -20,7 +20,7 @@ public static IEnumerable<TSource> Reverse<TSource>(this IEnumerable<TSource> so
             return new ReverseIterator<TSource>(source);
         }
 
-        private sealed class ReverseIterator<TSource> : Iterator<TSource>, IIListProvider<TSource>
+        private sealed class ReverseIterator<TSource> : Iterator<TSource>, IPartition<TSource>
```

Comment:
> Then they could do ToArray or ToList.

OK, you have convinced me; I additionally just did a grep search through Roslyn's `src/` and most of the usages are `foreach` or `ToArray`. I'll take this change out for now + reopen later if it turns out to be desirable.